### ISSUE TITLE
Order of resampled dataset is wrong

### ISSUE THREAD

**Issue by sayakpaul** [tensorflow/tensorflow#49007]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Colab
- TensorFlow installed from (source or binary): Binary
- TensorFlow version (use command below): 2.4.1

**Describe the current behavior**

I am using `tf.data.experimental.sample_from_datasets()` to balance a dataset. Now, I am creating two data loaders out of the resampled dataset and then zipping them up.  When I am iterating through the zipped dataset the images are coming in different orders. 

Let's walk this through step by step:

**Data**

```python
(x_train, y_train), (_, _) = tf.keras.datasets.cifar10.load_data()
sampled_idx = np.random.choice(len(x_train), 4000)
sampled_train, sampled_labels = x_train[sampled_idx], y_train[sampled_idx].squeeze()
support_ds = tf.data.Dataset.from_tensor_slices((sampled_train, sampled_labels))
```

**Utilities**

```python
def support_sampler(ds):
    ds_list = []
    for i in np.arange(0, 10):
        ds_label = ds.filter(lambda image, label: label == i).repeat()
        ds_list.append(ds_label)
    return ds_list

def get_support_ds(ds, bs=640):
    listed_ds = support_sampler(ds)
    balanced_ds = tf.data.experimental.sample_from_datasets(listed_ds, [0.1] * 10)

    loaders = tuple()
    for _ in range(2):
        loaders += (balanced_ds, )

    final_ds = tf.data.Dataset.zip(loaders)
    return final_ds.batch(bs)
```

**Issue verification**

Get a sample batch first:

```python
sample_support_ds = get_support_ds(support_ds)
support_images_one, support_images_two = next(iter(sample_support_ds))
```

**Plot the images from `support_images_one`**

```python
plt.figure(figsize=(7, 7))
for i, image in enumerate(support_images_one[0][:9]):
    ax = plt.subplot(3, 3, i + 1)
    plt.imshow(image.numpy().astype("int"))
    plt.title(int(support_images_one[1][i]))
    plt.axis("off")
```

Gives:

![image](https://user-images.githubusercontent.com/22957388/117529164-a54af980-aff3-11eb-9c8e-8e1312c2c38c.png)

The same thing with `support_images_two[0][:9]` (with the labels printed accordingly):

![image](https://user-images.githubusercontent.com/22957388/117529176-be53aa80-aff3-11eb-8c6f-2792b60b3bdb.png)

**Describe the expected behavior**

The image batches should be the same.

**[Contributing](https://www.tensorflow.org/community/contribute)** - Do you
want to contribute a PR? (yes/no): - Briefly describe your candidate solution
(if contributing): No

**Standalone code to reproduce the issue**

[Colab Notebook](https://colab.research.google.com/gist/sayakpaul/7e003afbd3714549d6ecb0f39b6b81b4/scratchpad.ipynb)

**Comment by sayakpaul** [comment]
Found out a solution. 

If we simply set the seed while doing the sampling step with `tf.data.experimental.sample_from_datasets()` this can be solved.
